### ISSUE TITLE
Retry functionality - additional toggles

### ISSUE THREAD

**Issue by wguo-github** [guregu/dynamo#222]
Hi, I noticed that the library provides in-built retry functionality (i.e. defined [here](https://github.com/guregu/dynamo/blob/master/retry.go#L28)). My understanding is that the retry capability is hard-coded and can't be disabled, nor can the maximum retries be configured in anyway - it will instead keep retrying with exponential backoff until the context deadline is exceeded (please correct me if I'm wrong). Would it be possible to add a parameter that allows the user to set the retry limit? Thanks!

**Comment by guregu** [comment]
Hello. Do you want to configure the # of tries or just disable it? Or both?

In general, retrying with exponential backoff is required for using the DynamoDB APIs as pretty much every request can fail with a "try again" error:
https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Programming.Errors.html#Programming.Errors.RetryAndBackoff

**Comment by guregu** [comment]
I think when this was first made the Go DynamoDB SDK didn't retry at all (but it was a long time ago, before aws-sdk-go v1), hence the retry code added here.
I think it would not be too hard to add some more config. We could probably just re-use the retrying interface from the official SDK (although I need to double-check the aws-sdk-go v2 interface to make sure it's similar).

**Comment by guregu** [comment]
@wguo-github Would #223 work for you?

---

#### Limiting or disabling retrying

The maximum number of retries can be configured via the `MaxRetries` field in the `*aws.Config` passed to `dynamo.New()`. A value of `0` will disable retrying. A value of `-1` means unlimited and is the default (however, context or `RetryTimeout` will still apply).

```go
db := dynamo.New(session, &aws.Config{
	MaxRetries: aws.Int(0), // disables automatic retrying
})
```

#### Custom retrying logic

If a custom [`request.Retryer`](https://pkg.go.dev/github.com/aws/aws-sdk-go/aws/request#Retryer) is set via the `Retryer` field in `*aws.Config`, dynamo will delegate retrying entirely to it, taking precedence over other retrying settings. This allows you to have full control over all aspects of retrying.

Example using [`client.DefaultRetryer`](https://pkg.go.dev/github.com/aws/aws-sdk-go/aws/client#DefaultRetryer):

```go
retryer := client.DefaultRetryer{
	NumMaxRetries:    10,
	MinThrottleDelay: 500 * time.Millisecond,
	MaxThrottleDelay: 30 * time.Second,
}
db := dynamo.New(session, &aws.Config{
	Retryer: retryer,
})
```

**Comment by wguo-github** [comment]
@guregu That looks perfect, thanks for making the change!

**Comment by guregu** [comment]
Awesome. I will merge & release soon.

**Comment by guregu** [comment]
Now released in [v1.21.0](https://github.com/guregu/dynamo/releases/tag/v1.21.0). Thank you for making the issue. Let me know if you run into any problems.
